### PR TITLE
Fixing file system watcher leak for PackageLag monitor.

### PR DESCRIPTION
**Pull request by ryuyu**
We were leaking file system watchers on every Init call, most likely due to event listeners causing GC to not clean them up.

This fixes that leak. This should be replicated to JsonConfigurationJob at some point, as JsonConfigurationJob base also has this issue.

@dtivel @chenriksson @joelverhagen @loic-sharma

### CODE REVIEW

File: src/PackageLagMonitor/Job.cs [hunk 1]
```diff
@@ -40,6 +40,7 @@ public class Job : JobBase
         private ICatalogClient _catalogClient;
         private IServiceProvider _serviceProvider;
         private PackageLagMonitorConfiguration _configuration;
+        private IConfigurationBuilder _builder = null;
```
Comment (dtivel) [inline src/PackageLagMonitor/Job.cs]: This is unused.

File: src/PackageLagMonitor/Job.cs [hunk 2]
```diff
@@ -60,19 +61,19 @@ private IConfigurationRoot GetConfigurationRoot(string configurationFilename)
             Logger.LogInformation("Using the {ConfigurationFilename} configuration file", configurationFilename);
             var builder = new ConfigurationBuilder()
                 .SetBasePath(Environment.CurrentDirectory)
-                .AddJsonFile(configurationFilename, optional: false, reloadOnChange: true);
+                .AddJsonFile(configurationFilename, optional: false, reloadOnChange: false);
```
Comment (dtivel) [inline src/PackageLagMonitor/Job.cs]: Assuming that GitHub's search index is current and complete, there are 2 other places ([here](https://github.com/NuGet/NuGet.Jobs/blob/352ce8d5a2306ce735703e2fa2bc48ebe82ce24b/src/NuGet.Jobs.Common/JsonConfigurationJob.cs#L79) and [here](https://github.com/NuGet/NuGet.Jobs/blob/47326c56f822f18e09bf203fda7770372754bd23/src/NuGet.Services.Validation.Orchestrator/Job.cs#L127)) that pass `true` for `reloadOnChange`.  I realize that this PR targets an issue with the Monitoring.PackageLag job, but does this issue exist in either of those jobs as well?

Comment (ryuyu) [reply src/PackageLagMonitor/Job.cs]: Yes it is likely the issue exists in the other locations as well. I will update it in those locations as well.

File: src/PackageLagMonitor/Job.cs (hunk 1 above)
Comment (ryuyu) [reply src/PackageLagMonitor/Job.cs]: Good catch. Change from a different solution i explored. Deleted.

File: src/PackageLagMonitor/Job.cs [hunk 3]
```diff
@@ -60,19 +60,19 @@ private IConfigurationRoot GetConfigurationRoot(string configurationFilename)
...
 
             var uninjectedConfiguration = builder.Build();
 
             var secretReaderFactory = new ConfigurationRootSecretReaderFactory(uninjectedConfiguration);
             var cachingSecretReaderFactory = new CachingSecretReaderFactory(secretReaderFactory, KeyVaultSecretCachingTimeout);
             var secretInjector = cachingSecretReaderFactory.CreateSecretInjector(cachingSecretReaderFactory.CreateSecretReader());
 
-            builder = new ConfigurationBuilder()
+            var injectedBuilder = new ConfigurationBuilder()
```
Comment (joelverhagen) [inline src/PackageLagMonitor/Job.cs]: Is this change necessary? The other spots seem to still reassign the `builder` local.

**Comment by joelverhagen** [comment]
Have we manually tested that secrets are still reloaded? I think it should be fine, but this is the one "reload" case that we really care about.

File: src/PackageLagMonitor/Job.cs (hunk 3 above)
Comment (ryuyu) [reply src/PackageLagMonitor/Job.cs]: Doesn't look like. I believe it might have been a left over from testing a different fix. I'll revert.

**Comment by ryuyu** [comment]
I tested that secrets reload manually. 👍
